### ISSUE TITLE
guard-for-in is unable to detect if statement guarding for...in loop

### ISSUE THREAD

**Issue by LJNeon** [eslint/eslint#10902]
**Tell us about your environment**
* **ESLint Version:** 5.6.1
* **Node Version:** 10.11.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
`default`
**Please show your full configuration:**
<details>
<summary>Configuration</summary>
(our actual config is over 200 lines but this is the only relevant part)

```yml
env: {
  es6: true,
  node: true
}
rules: {
  guard-for-in: 2
}
parserOptions: {
  ecmaVersion: 2018
}
```
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
for(const col in xp) {
  if(!xp.hasOwnProperty(col))
    return;
  // ...
}
```
```bash
eslint . --ext .js
```
**What did you expect to happen?**
Eslint would consider the above code valid as we do guard the for...in loop.
**What actually happened? Please include the actual, raw output from ESLint.**
Eslint was unable to detect the if statement that guards that for...in loop.
```bash
/../src/services/database.js
  66:5  error  The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype  guard-for-in
```

**Comment by g-plane** [comment]
I *cannot* reproduce: https://eslint.org/demo/#eyJ0ZXh0IjoiZnVuY3Rpb24gZm4oKSB7XHJcbmZvcihjb25zdCBjb2wgaW4geHApIHtcclxuICBpZigheHAuaGFzT3duUHJvcGVydHkoY29sKSlcclxuICAgIHJldHVybjtcclxuICAvLyAuLi5cclxufVxyXG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7Imd1YXJkLWZvci1pbiI6Mn0sImVudiI6e319fQ==

**Comment by ilyavolodin** [comment]
@LJNeon Could you confirm that your rule is configured correctly? It doesn't seem like this is reproducible in our online demo (which always runs latest version of ESLint). Thanks @g-plane for repro link.

**Comment by not-an-aardvark** [comment]
The issue wasn't reproducible bevause the code sample didn't contain any other statements after the `if`. I could reproduce the reported problems in the demo [here](https://eslint.org/demo/#eyJ0ZXh0IjoiZnVuY3Rpb24gZm4oKSB7XHJcbmZvcihjb25zdCBjb2wgaW4geHApIHtcclxuICBpZigheHAuaGFzT3duUHJvcGVydHkoY29sKSlcclxuICAgIHJldHVybjtcclxuICBzb21ldGhpbmdFbHNlKClcclxufVxyXG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7Imd1YXJkLWZvci1pbiI6Mn0sImVudiI6e319fQ==).

The current rule avoids reporting an error when `continue` is found in an if-statement at the start of the loop, but it doesn't check for `return`. Are you sure you meant to use `return` here? It seems like one would typically want to skip properties on the prototype rather than exiting the entire function when such a property is encountered.

**Comment by LJNeon** [comment]
It looks like in our use case `return` worked as intended but using `continue` is preferable and more correct. I'll close this issue as changing it to `continue` fixed our problem.
